### PR TITLE
Iox #743 Fix website generation and typos

### PR DESCRIPTION
**Pull request by mossmaurice**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

## Notes for Reviewer
* Fix include issue with MkDocs
* Fix typos
* Update to mike v1.0.0 to avoid HTML duplication
* Fix links to methods in doxygen docu

**These changes will be backported to `release_1.0` with the next v1.x release.**

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes partly #743

### CODE REVIEW

File: doc/website/advanced/iceoryx_utils.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ---
 title: Safe building blocks
 ---
-{! ./../iceoryx_utils/README.md !}
+{! ../iceoryx/iceoryx_utils/README.md !}
```
Comment (elfenpiff) [inline doc/website/advanced/iceoryx_utils.md:4]: How can a reviewer ensure that the relative links are correct now? Where can I see where the starting point for this relative link is?

File: doc/website/index.md [hunk 2]
```diff
@@ -8,9 +8,9 @@ title: Home
         width="50%">
 </p>
 
-Eclipse iceoryx™ is an inter-process-communication midddleware that enables virtually limitless data transmissions at constant time.
+Eclipse iceoryx™ is an inter-process-communication middleware that enables virtually limitless data transmissions at constant time.
```
Comment (elfenpiff) [inline doc/website/index.md:11]: Ouh ... only two d's I thought the third d was to there to praise hypnotoad

File: doc/website/advanced/iceoryx_utils.md (hunk 1 above)
Comment (mossmaurice) [reply doc/website/advanced/iceoryx_utils.md:4]: By running either `mkdocs serve` or `mike deploy`, see [export script](https://github.com/eclipse-iceoryx/iceoryx/blob/master/tools/export-docu-to-website.sh#L74-L86).

If no [`base_path`](https://github.com/cmacmackin/markdown-include/#configuration) is specified when using `markdown-include`...

> Default: the run-directory

**Review by dkroenke — DISMISSED** [review]

File: doc/website/advanced/iceoryx_utils.md (hunk 1 above)
Comment (mossmaurice) [reply doc/website/advanced/iceoryx_utils.md:4]: I've added a sanity check in the export script.

**Review by dkroenke — APPROVED** [review]

**Review by budrus — APPROVED** [review]

**Pull request merged**